llvm/include/llvm/ProfileData/MemProf.h

```diff
@@ -561,19 +488,21 @@ class RecordLookupTrait {
 
   data_type ReadData(uint64_t K, const unsigned char *D,
                      offset_type /*Unused*/) {
-    Record = IndexedMemProfRecord::deserialize(Schema, D, Version1);
+    Record = IndexedMemProfRecord::deserialize(Schema, D, Version);
     return Record;
   }
 
 private:
+  // Holds the MemProf version.
+  IndexedVersion Version;
   // Holds the memprof schema used to deserialize records.
   MemProfSchema Schema;
   // Holds the records from one function deserialized from the indexed format.
   IndexedMemProfRecord Record;
 };
 
 // Trait for writing IndexedMemProfRecord data to the on-disk hash table.
-class RecordWriterTrait {
+template <IndexedVersion Version> class RecordWriterTrait {
```

Comment:
Why pass in the version to RecordLookupTrait but template RecordWriterTrait? If it is possible to make them consistent that would be great but I might be missing the reasoning.

Reply:
That is because `EmitKeyDataLength` is a static function, so I wasn't able to make `Version` a member variable (as opposed to a template parameter).

Now, on a deeper look, I may be able to make `EmitKeyDataLength` a non-static member function.  I'll see if I can address that in a follow-up patch.